### ISSUE TITLE
Increase AWS limits in some regions

### ISSUE THREAD

**Issue by rifelpet** [kubernetes/k8s.io#1921]
I can't confirm this because I do not have access to the AWS accounts used by prow jobs, but I believe that some AWS regions have lower resource limits than others.

Example VPC limits being reached in [ap-southeast-1a](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/directory/pull-kops-e2e-kubernetes-aws/1381638281560592384#1:build-log.txt%3A569), [eu-central-1a](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/directory/pull-kops-e2e-kubernetes-aws/1381463978663743488#1:build-log.txt%3A494), [eu-north-1a](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/directory/pull-kops-e2e-kubernetes-aws/1381454454682816512#1:build-log.txt%3A487), and [eu-west-1a](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/directory/pull-kops-e2e-kubernetes-aws/1381436067751661568#1:build-log.txt%3A504).

`W0412 16:10:28.895604   10570 executor.go:139] error running task "VPC/e2e-be8b7a2dfe-ff1eb.test-cncf-aws.k8s.io" (5m18s remaining to succeed): error creating VPC: VpcLimitExceeded: The maximum number of VPCs has been reached.`

I know that in the past we have increased the service limits in some regions and zones but it would be nice to have those limits be increased consistently. New regions will be added, new AWS accounts will be added; there are many opportunities for these limits to drift. Tools like [cloud custodian](https://cloudcustodian.io/docs/aws/examples/accountservicelimit.html) can increase limits automatically.

/cc @justinsb

**Comment by ameukam** [comment]
@idvoretskyi Any help possible ? :-)

**Comment by idvoretskyi** [comment]
cc @justinsb - is this something you can take care of?

**Comment by spiffxp** [comment]
/sig cluster-lifecycle
/sig cloud-provider
/area provider/aws
a guess at labels for the owning sigs/subprojects

/area cluster-infra
k8s-infra specific label (maybe we should have infra/aws and infra/gcp labels?)

/assign @justinsb @idvoretskyi @thockin 
Assigning to k8s-infra-aws-admins

**Comment by sedefsavas** [comment]
+100 for increasing the quotas.

cluster-api-provider-aws project tests take a long time, and having very limited quotas on services does not allow us gain much by running the tests in parallel. Service quotas for especially elastic IPs, VPCs, nat gateways etc. are in general limited for our tests.

cc @randomvariable

**Comment by randomvariable** [comment]
related to https://github.com/kubernetes/k8s.io/pull/1839

**Comment by idvoretskyi** [comment]
@rifelpet which AWS account(s) are we speaking about (we have multiple)?

**Comment by rifelpet** [comment]
@idvoretskyi I'm experiencing this with Kops on 768319786644, but cluster-api-provider-aws runs on different accounts

**Comment by sedefsavas** [comment]
`cluster-api-provider-aws` uses the same account with Kops by setting  `preset-aws-credential: "true"`

We observe the following quotas for `us-west-2` region:
```
classiclb: 20
ec2: 512
eip: 5
igw: 5
ngw: 5
vpc: 5
```

We are facing bottleneck with especially EIP and VPC limits.

**Comment by sedefsavas** [comment]
@idvoretskyi 
What are the steps we can take to proceed on this issue? Do you know who to reach out?

**Comment by idvoretskyi** [comment]
@rifelpet this is not an account I have access to, so I believe someone from the CAPA team has to handle this.

**Comment by randomvariable** [comment]
I think it's @justinsb for these.

Kops and CAPA are using different sets of accounts at the moment I believe.

**Comment by idvoretskyi** [comment]
I believe the limits have already been increased, @rifelpet can you please check?

**Comment by randomvariable** [comment]
I may have managed to get some service quotas raised asynchronously through test runs, but the ones we're after for Cluster API are (and all for us-west-2):

```go
	serviceQuotas["igw"] = &ServiceQuota{
		ServiceCode:         "vpc",
		QuotaName:           "Internet gateways per Region",
		QuotaCode:           "L-A4707A72",
		DesiredMinimumValue: 20,
	}

	serviceQuotas["ngw"] = &ServiceQuota{
		ServiceCode:         "vpc",
		QuotaName:           "NAT gateways per Availability Zone",
		QuotaCode:           "L-FE5A380F",
		DesiredMinimumValue: 20,
	}

	serviceQuotas["vpc"] = &ServiceQuota{
		ServiceCode:         "vpc",
		QuotaName:           "VPCs per Region",
		QuotaCode:           "L-F678F1CE",
		DesiredMinimumValue: 20,
	}

	serviceQuotas["ec2"] = &ServiceQuota{
		ServiceCode:         "ec2",
		QuotaName:           "Running On-Demand Standard (A, C, D, H, I, M, R, T, Z) instances",
		QuotaCode:           "L-1216C47A",
		DesiredMinimumValue: 20,
	}

	serviceQuotas["eip"] = &ServiceQuota{
		ServiceCode:         "ec2",
		QuotaName:           "EC2-VPC Elastic IPs",
		QuotaCode:           "L-0263D0A3",
		DesiredMinimumValue: 100,
	}

	serviceQuotas["classiclb"] = &ServiceQuota{
		ServiceCode:         "elasticloadbalancing",
		QuotaName:           "Classic Load Balancers per Region",
		QuotaCode:           "L-E9E9831D",
		DesiredMinimumValue: 20,
	}
```

**Comment by ameukam** [comment]
@rifelpet @randomvariable Do you still need assistance about this ? If it's solved what the escalation process ?

**Comment by rifelpet** [comment]
@ameukam We've increased the limits and are no longer running into quotas, but it would be nice to have a formal strategy for dealing with this in the future, for example if we expand into new regions or start using new resource types. If you think that discussion should be its own issue I'm fine with closing this one out.

**Comment by sedefsavas** [comment]
We observe a reduction in ec2 quota from 512 to 32 which is causing our tests to fail.
Before [this](https://storage.googleapis.com/kubernetes-jenkins/logs/periodic-cluster-api-provider-aws-e2e/1464065565801844736/artifacts/initial-resource-quotas.yaml) --> now [this](https://storage.googleapis.com/kubernetes-jenkins/logs/periodic-cluster-api-provider-aws-e2e/1469140416698781696/artifacts/initial-resource-quotas.yaml)
What might have reduced the quotas?

**Comment by ameukam** [comment]
> @ameukam We've increased the limits and are no longer running into quotas, but it would be nice to have a formal strategy for dealing with this in the future, for example if we expand into new regions or start using new resource types. If you think that discussion should be its own issue I'm fine with closing this one out.

@rifelpet I think it's better to open a new one about AWS quotas so we tag the right people. Thanks!

**Comment by eddiezane** [comment]
This should be sorted.

/close
